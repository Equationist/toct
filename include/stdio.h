/* stdio.h - Standard Input/Output Library for CompilerKit C Compiler */

#ifndef _STDIO_H
#define _STDIO_H

/* Standard I/O types */
typedef struct {
    int _fd;
    int _flags;
    char* _buffer;
} FILE;

/* Standard streams */
extern FILE* stdin;
extern FILE* stdout; 
extern FILE* stderr;

/* Standard I/O functions */
int printf(const char* format, ...);
int fprintf(FILE* stream, const char* format, ...);
int sprintf(char* str, const char* format, ...);
int snprintf(char* str, int size, const char* format, ...);

int scanf(const char* format, ...);
int fscanf(FILE* stream, const char* format, ...);
int sscanf(const char* str, const char* format, ...);

int putchar(int c);
int puts(const char* s);
int fputs(const char* s, FILE* stream);
int fputc(int c, FILE* stream);

int getchar(void);
char* gets(char* s);
char* fgets(char* s, int size, FILE* stream);
int fgetc(FILE* stream);
int ungetc(int c, FILE* stream);

FILE* fopen(const char* filename, const char* mode);
FILE* freopen(const char* filename, const char* mode, FILE* stream);
int fclose(FILE* stream);
int fflush(FILE* stream);

long ftell(FILE* stream);
int fseek(FILE* stream, long offset, int whence);
void rewind(FILE* stream);
int feof(FILE* stream);
int ferror(FILE* stream);
void clearerr(FILE* stream);

/* File position constants */
#define SEEK_SET 0
#define SEEK_CUR 1
#define SEEK_END 2

/* End of file constant */
#define EOF (-1)

/* Buffer size */
#define BUFSIZ 8192

/* Null pointer */
#ifndef NULL
#define NULL ((void*)0)
#endif

#endif /* _STDIO_H */