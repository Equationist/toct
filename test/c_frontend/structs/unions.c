/* Test union types */
#include <stdio.h>

union Data {
    int i;
    float f;
    char c;
};

int main() {
    union Data data;
    
    data.i = 42;
    printf("data.i = %d\n", data.i);
    
    data.f = 3.14f;
    printf("data.f = %.2f\n", data.f);
    printf("data.i after setting f = %d\n", data.i);
    
    data.c = 'A';
    printf("data.c = %c\n", data.c);
    printf("data.c as int = %d\n", (int)data.c);
    
    printf("sizeof(union Data) = %zu\n", sizeof(union Data));
    
    return 0;
}