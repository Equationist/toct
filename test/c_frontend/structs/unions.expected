data.i = 42
data.f = 3.14
data.i after setting f = 1078523331
data.c = A
data.c as int = 65
sizeof(union Data) = 4